src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -430,14 +430,20 @@ public async Task<IActionResult> GetTriggers()
         {
             _metricsLogger.LogEvent(MetricEventNames.GetTriggersInvoked);
 
-            var result = await _functionsSyncManager.GetTriggersAsync();
+            // This endpoint is only available in validation mode
+            if (_environment.IsInValidationMode())
+            {
+                var result = await _functionsSyncManager.GetTriggersAsync();
 
-            // GetTriggersAsync() does not swallow exceptions,so any failures will still be a 500.
-            // The only result.Success == false we need to consideris when the environment
-            // does not support sync triggers.
-            return result.Success
-                ? Ok(result)
-                : StatusCode(StatusCodes.Status403Forbidden, new { status = result.Error });
+                // GetTriggersAsync() does not swallow exceptions,so any failures will still be a 500.
+                // The only result.Success == false we need to consideris when the environment
+                // does not support sync triggers.
+                return result.Success
+                    ? Ok(result)
+                    : StatusCode(StatusCodes.Status403Forbidden, new { status = result.Error });
+            }
+
+            return NotFound();
```

Comment:
Hmmm, why not just also return 403 in this case? The endpoint does exist (so not a 404), however just as above due to host conditions we don't allow the call.

Reply:
Main reason is to differentiate between:
case 1: it is not IsSyncTriggersEnvironment. This will result in 403.
case 2: it is not a validation mode, so the API itself is not supported. I'm open for any other suggestions if not 404 :)

Reply:
Why do we need to differentiate? FDM is the only caller of this API - why do you need to differentiate? In my mind, both of these cases are just saying that the host isn't in a state that permits this API, so we return 403 Forbidden.

Reply:
you are right. We can say: it is forbidden if not validation mode. Updated the status code. Thank you!